pandas/core/ops/__init__.py

```diff
@@ -428,6 +429,7 @@ def f(self, other, axis=default_axis, level=None, fill_value=None):
 
         axis = self._get_axis_number(axis) if axis is not None else 1
 
+        other = maybe_prepare_scalar_for_op(other, (self.shape[axis],))
```

Comment:
why self.shape[axis] instead of self.shape?

Reply:
Because `self` is a dataframe, and we want a 1D shape (rows or columns depending on the `axis`)

Reply:
passing shape only matters when we have a scalar, in which case broadcasting to self.shape is simpler

Reply:
OK, changed to self.shape